backend/de.metas.business/src/main/java/de/metas/inout/impl/InOutBL.java

```diff
@@ -663,6 +669,10 @@ public CurrencyConversionContext getCurrencyConversionContext(@NonNull final I_M
 		final ForexContractRef forexContractRef = InOutDAO.extractForeignContractRef(inout);
 		if (forexContractRef != null)
 		{
+			Optional.ofNullable(forexContractRef.getForexContractId())
```

Comment:
loud thinking: why validate it here and why not when the inout is generated?

Reply:
bc I want to make sure the section code is right when the document gets posted, not created.. as the section code can be changed at any given time.